bindings/python/bt2/bt2/internal/fields.py

```diff
@@ -0,0 +1,678 @@
+        self.value = self & other
+        return self
+
+    def __ixor__(self, other):
+        self.value = self ^ other
+        return self
+
+    def __ior__(self, other):
+        self.value = self | other
+        return self
+
+
+class _RealField(_NumericField, numbers.Real):
+    pass
+
+
+class _IntegerField(_IntegralField):
+    _NAME = 'Integer'
+
+    def _value_to_int(self, value):
+        if not isinstance(value, numbers.Real):
+            raise TypeError('expecting a real number object')
+
+        value = int(value)
+
+        if self.field_type.is_signed:
+            utils._check_int64(value)
+        else:
+            utils._check_uint64(value)
+
+        return value
+
+    @property
+    def _value(self):
+        if self.field_type.is_signed:
+            ret, value = self._Domain.field_integer_signed_get_value(self._ptr)
+        else:
+            ret, value = self._Domain.field_integer_unsigned_get_value(self._ptr)
+
+        if ret < 0:
+            utils._handle_ret(ret, "cannot get integer field's value")
+
+        return value
+
+    def _set_value(self, value):
+        value = self._value_to_int(value)
+
+        if self.field_type.is_signed:
+            ret = self._Domain.field_integer_signed_set_value(self._ptr, value)
+        else:
+            ret = self._Domain.field_integer_unsigned_set_value(self._ptr, value)
+
+        utils._handle_ret(ret, "cannot set integer field object's value")
+
+    value = property(fset=_set_value)
+
+
+class _FloatingPointNumberField(_RealField):
+    _NAME = 'Floating point number'
+
+    def _value_to_float(self, value):
+        if not isinstance(value, numbers.Real):
+            raise TypeError("expecting a real number object")
+
+        return float(value)
+
+    @property
+    def _value(self):
+        ret, value = self._Domain.field_floating_point_get_value(self._ptr)
+
+        if ret < 0:
+            utils._handle_ret(ret, "cannot get floating point number field's value")
+
+        return value
+
+    def _set_value(self, value):
+        value = self._value_to_float(value)
+        ret = self._Domain.field_floating_point_set_value(self._ptr, value)
+        utils._handle_ret(ret, "cannot set floating point number field object's value")
+
+    value = property(fset=_set_value)
+
+
+class _EnumerationField(_IntegerField):
+    _NAME = 'Enumeration'
+
+    @property
+    def integer_field(self):
+        int_field_ptr = self._Domain.field_enumeration_get_container(self._ptr)
+        assert(int_field_ptr)
+        return self._Domain.create_field_from_ptr(int_field_ptr)
+
+    def _set_value(self, value):
+        self.integer_field.value = value
+
+    def _repr(self):
+        labels = [repr(v.name) for v in self.mappings]
+        return '{} ({})'.format(self._value, ', '.join(labels))
+
+    value = property(fset=_set_value)
+
+    @property
+    def _value(self):
+        return self.integer_field._value
+
+    @property
+    def mappings(self):
```

Comment:
Not available in CTF writer.

Reply:
What do we do with the _repr function that uses it?

Reply:
Changed the _repr function to get the labels that matches from the field_type directly.